### PR TITLE
Cu 86b0x36kf level system

### PR DESCRIPTION
**Pull request by ReinhardtBotha**
When you enable Realtime Tracking the level system will start to update in the front end. 

At the moment:
- you will lose 1 hp every second if the posture is bad or gain 1 hp for every good posture.
- When the HP is 0, it will stop increasing XP.
- The XP will increase by 10XP every 5 seconds if your posture is good.
- It will lookup the values in levels.ts and level up based on that (currently level 2 at 20XP, level 3 at 40XP).

The values are just for testing purposes at the moment.

TODO: backend and database integration

### CODE REVIEW

**Comment by zeckkimon** [comment]
Task linked: [CU-86b0x36kf Level System](https://app.clickup.com/t/86b0x36kf)

**Comment by zeckkimon** [comment]
Task linked: [CU-86b0x36t4 Polish the Tracking Behaviour](https://app.clickup.com/t/86b0x36t4)

**Comment by whamester** [comment]
> * When the XP is 0, it will stop increasing XP.

Did you mean when the HP is 0, it will stop increasing the XP?

File: src/components/scoring/XPSystem.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { useEffect } from "react";
+import { useUser } from "@src/state/useUser";
+
+const XPSystem = () => {
+  const userHP = useUser((state) => state.user.hp);
+  const currentPosture = useUser((state) => state.currentPosture);
+  const setXP = useUser((state) => state.setXP);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
```
Comment (whamester) [inline src/components/scoring/XPSystem.tsx]: you might want to check if the tracking is enabled to activate the interval, and also clear the interval when the real-time tracking is disabled.

another way could be to check if the postureData length is multiple of 5, so every 5 readings you check the currentPosture, the postureData list will no be updated if the isTrackingEnabled flag is false.

**Review by whamester — APPROVED** [review]

**Review by Victorportus — APPROVED** [review]

**Review by zeckkimon — APPROVED** [review]

**Comment by ReinhardtBotha** [comment]
> > * When the XP is 0, it will stop increasing XP.
> 
> Did you mean when the HP is 0, it will stop increasing the XP?

Yes sorry that was a typo. I updated the description now. Thanks!

File: src/components/scoring/XPSystem.tsx (hunk 1 above)
Comment (ReinhardtBotha) [reply src/components/scoring/XPSystem.tsx]: Thank you for this, I like the second approach more actually! Implemented here: https://github.com/LangaraBeetles/weaup-react-native-app/pull/31/commits/2f7d0facf9c959a4e9753fc9365c44f4d5a3dc65

**Review by whamester — APPROVED** [review]

**Pull request merged**